src/Symfony/Component/HttpKernel/Profiler/Profiler.php

```diff
@@ -178,6 +178,10 @@ public function collect(Request $request, Response $response, \Exception $except
             $profile->setIp('Unknown');
         }
 
+        if ($prevToken = $response->headers->get('X-Debug-Token')) {
```

Comment:
Doesn't this make any sub request a forwarded one?

Reply:
Yes :) basically if the response has a x-debug-token header _already_, we assume a forwarded request. Thats the trick here.

Reply:
And thus if we have a `X-Previous-Debug-Token` in RequestDataCollector we know it was forwarded (it had a x-debug-token previously).